wp-admin/js/customize-controls-site-icon-maskable.js

```diff
@@ -28,8 +29,59 @@ wp.customize(
 				// Set initial active state.
 				updateActive();
 
+				/**
+				 * Validate site icons for its presence and size.
+				 */
+				const iconValidation = () => {
+					const iconData = wp.customize
+						.control('site_icon')
+						.container.find('img.app-icon-preview');
+					const NotificationData = {
+						dismissible: true,
+						message: '',
+						type: 'error',
+						code: null,
+					};
+					const notifications = [];
+
+					if (!iconData.length) {
+						notifications.push(
+							new wp.customize.Notification('pwa_icon_not_set', {
+								...NotificationData,
+								message: PWA_IconMessages.pwa_icon_not_set,
+							})
+						);
+					}
+
+					if (
+						iconData.length &&
+						(iconData[0].naturalHeight < 512 ||
+							iconData[0].naturalHeight < 512)
+					) {
+						notifications.push(
+							new wp.customize.Notification(
+								'pwa_icon_too_small',
+								{
+									...NotificationData,
+									message:
+										PWA_IconMessages.pwa_icon_too_small,
+								}
+							)
+						);
+					}
+
+					notifications.map((notification) => {
+						wp.customize
+							.section('title_tagline')
+							.notifications.add(notification);
+
+						return notification;
+					});
+				};
+
 				// Update active state whenever the site_icon setting changes.
-				siteIconSetting.bind(updateActive);
+				// Update notification when site_icon setting changes.
+				siteIconSetting.bind(updateActive).bind(iconValidation);
 
```

Comment:
We're missing a `iconValidation()` call here to set the initial state:

```suggestion
				iconValidation();

```